### PR TITLE
malt: new package (versions : 1.2.1)

### PR DESCRIPTION
**Pull request by svalat**
Add support for the MALT memory profiling tool :
 - https://memtt.github.io/malt
 - https://github.com/memtt/malt

**Variants**
 - MALT provide an optional support of a webapp via qt instead of using a distinct webserver + browser. This is enabled with `+qt`.
 - MALT is delivers as an instrumentation backend which is a .so file and the webview based on nodejs which is enabled by default. It can be disabled with variants not to fetch nodejs by using `-nodejs`.

**Version**
The previous versions had issues which prevent from use so I added only the last one : 1.2.1. 
I Don't know what is the policy you prefer to follow : 
 - list all
 - keep only stables.

**Notice**: 
I'm the creator and maintainer of this open source software.

### CODE REVIEW

File: var/spack/repos/builtin/packages/malt/package.py
Comment (besnardjb) [inline var/spack/repos/builtin/packages/malt/package.py]: Dependencies are by default of type (build, link) (see https://spack.readthedocs.io/en/latest/packaging_guide.html#dependency-types) however I suppose you use `addr2line` for symbol resolution only at runtime, should you add?

```suggestion
    depends_on("binutils", type="run")
```

File: var/spack/repos/builtin/packages/malt/package.py
Comment (besnardjb) [inline var/spack/repos/builtin/packages/malt/package.py]: Usualy passed as named argument.

```suggestion
    depends_on("qt", when="+qt")
```

File: var/spack/repos/builtin/packages/malt/package.py
Comment (besnardjb) [inline var/spack/repos/builtin/packages/malt/package.py]: Dependencies are by default of type (build, link) (see https://spack.readthedocs.io/en/latest/packaging_guide.html#dependency-types) , you will not have the node-js interpreter in path if you do not put a type.

Also you may consider fixing a minimal node-js version as support level changes.

Left build below as you may have some scripts at install ?

```suggestion
    depends_on("node-js@18:", when="+nodejs", type=("build", "run"))
```

**Review by besnardjb — COMMENTED** [review]
Tested it with my changes and compiled.

File: var/spack/repos/builtin/packages/malt/package.py
Comment (besnardjb) [reply var/spack/repos/builtin/packages/malt/package.py]: It seems it works without "run" not sure why ;-)

File: var/spack/repos/builtin/packages/malt/package.py
Comment (svalat) [reply var/spack/repos/builtin/packages/malt/package.py]: OK, I fixed. Thanks ;).

File: var/spack/repos/builtin/packages/malt/package.py
Comment (svalat) [reply var/spack/repos/builtin/packages/malt/package.py]: Fixed

File: var/spack/repos/builtin/packages/malt/package.py
Comment (svalat) [reply var/spack/repos/builtin/packages/malt/package.py]: Fixed

File: var/spack/repos/builtin/packages/malt/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/malt/package.py]: If you aren't adding arguments at this time, you can remove this method at this point since that is the default implementation result.  See the implementation at https://github.com/spack/spack/blob/383a343412216ad4f470da9bf1b6b96529cd1060/lib/spack/spack/build_systems/cmake.py#L389.

File: var/spack/repos/builtin/packages/malt/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/malt/package.py]: This method is not needed since the current `url` property will cover any version that adheres to the pattern here.  More at https://spack.readthedocs.io/en/latest/packaging_guide.html#version-urls.

**Review by tldahlgren — COMMENTED** [review]
Confirmed the sha256 but there's extra code that doesn't need to be in the package (at this point).

File: var/spack/repos/builtin/packages/malt/package.py
Comment (svalat) [reply var/spack/repos/builtin/packages/malt/package.py]: Fixed

File: var/spack/repos/builtin/packages/malt/package.py
Comment (svalat) [reply var/spack/repos/builtin/packages/malt/package.py]: Fixed

File: var/spack/repos/builtin/packages/malt/package.py
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/malt/package.py]: @besnardjb Great catch BTW!

File: var/spack/repos/builtin/packages/malt/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/malt/package.py]: Nit:

```suggestion
        description="Build the viewer NodeJS + QT web toolkit (requires NodeJS too)",
```

Comment (svalat) [reply var/spack/repos/builtin/packages/malt/package.py]: Fixed.

**Review by tldahlgren — DISMISSED** [review]
LGTM

**Comment by tldahlgren** [comment]
Will give @besnardjb a chance to comment before merging.

**Comment by besnardjb** [comment]
Looks perfect -- a beautiful compact recipe. Happy to have MALT in spack thanks @svalat!

File: var/spack/repos/builtin/packages/malt/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/malt/package.py]: Missed this.  The syntax for this changed recently (https://spack.readthedocs.io/en/latest/packaging_guide.html#maintainers).  A quick change and I'll merge it.

```suggestion
    maintainers("svalat")
```

**Comment by tldahlgren** [comment]
As you can see I took the liberty of updating the `maintainers` syntax.

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**
